DevTools/HackStudio/Editor.cpp

```diff
@@ -473,7 +473,7 @@ void Editor::set_document(GUI::TextDocument& doc)
     switch (code_document.language()) {
     case Language::Cpp:
         set_syntax_highlighter(make<GUI::CppSyntaxHighlighter>());
-        cpp_Language_server_connection().post_message(Messages::CppLanguageServer::FileOpened(code_document.file_path().string()));
+        m_language_client = create_language_client<LanguageClients::Cpp::ServerConnection>(project().root_directory());
```

Comment:
Wouldn't this spawn a new language server process whenever we open a file?
Since a language server is supposed to handle multiple files, I think it would make more sense to have a single instance of the language server that serves the whole project (like we had before), with editors sharing the connection to it.

Also, previously we spawned a C++ language server on HackStudio startup regardless of whether there are even .cpp files in the project. We could improve on that by looking at the `.hsp` file for knowing which language servers to spawn (alternatively we could spawn language servers lazily as needed).

Reply:
Hmm...I didn't think of that.
I'll make it share the servers.

> Lazily as needed

This is doing that though?

Reply:
> This is doing that though?

Oh, I meant that e.g when we open a .sh file for the first time in a project we would spawn a shell language server, and when we open another .sh file we would use the connection to the already running shell language server.

Maybe we could have a global map between HackStudio::Language and a language server connection?

Reply:
> Maybe we could have a global map between HackStudio::Language and a language server connection

Just a static local map per `ServerConnection` now.